src/instructlab/model/evaluate.py

```diff
@@ -0,0 +1,504 @@
+        if scores is None:
+            qna_to_scores[qna_file] = [score]
+        else:
+            scores.append(score)
+    qna_to_avg_scores = {}
+    for qna, scores in qna_to_scores.items():
+        qna_to_avg_scores[qna] = sum(scores) / len(scores)
+    return qna_to_avg_scores
+
+
+@click.command()
+@click.option(
+    "--model",
+    type=click.STRING,
+    help="Model to be evaluated - can be a local path or the name of a Hugging Face repository",
+)
+@click.option(
+    "--base-model",
+    type=click.STRING,
+    help="Base model to compare with 'model' for mt_bench_branch and mmlu_branch - can be a local path or the name of a Hugging Face repository",
+)
+@click.option(
+    "--benchmark",
+    type=click.Choice(list(BENCHMARK_TO_CLASS_MAP.keys())),
+    # case_sensitive=False,
+    help="Benchmarks to run during evaluation",
+)
+@click.option(
+    "--judge-model",
+    type=click.STRING,
+    help="Model to be used as a judge for running mt_bench or mt_bench_branch - can be a local path or the name of a Hugging Face repository",
+)
+@click.option(
+    "--output-dir",
+    type=click.Path(),
+    help="The directory to use for evaluation output from mt_bench or mt_bench_branch",
+)
+@click.option(
+    "--max-workers",
+    type=click.INT,
+    help="Max parallel workers to run the evaluation with for mt_bench or mt_bench_branch",
+)
+@click.option(
+    "--taxonomy-path",
+    type=click.Path(),
+    help="Taxonomy git repo path for running mt_bench_branch",
+)
+@click.option(
+    "--branch",
+    type=click.STRING,
+    help="Branch of taxonomy repo to eval QNAs against model",
+)
+@click.option(
+    "--base-branch",
+    type=click.STRING,
+    help="Base branch of taxonomy repo to eval QNAs against model for mt_bench_branch",
+)
+@click.option(
+    "--few-shots",
+    type=click.INT,
+    help="Number of examples. Needed for running mmlu or mmlu_branch.",
+)
+@click.option(
+    "--batch-size",
+    type=click.INT,
+    help="Number of GPUs. Needed for running mmlu or mmlu_branch.",
+)
+@click.option(
+    "--sdg-path",
+    type=click.Path(),
+    help="Path where all the MMLU Branch tasks are stored. Needed for running mmlu_branch.",
+)
+@click.pass_context
+def evaluate(
+    ctx,  # pylint: disable=unused-argument
+    model,
+    base_model,
+    benchmark,
+    judge_model,
+    output_dir,
+    max_workers,
+    taxonomy_path,
+    branch,
+    base_branch,
+    few_shots,
+    batch_size,
+    sdg_path,
+):
+    # get appropriate evaluator class from Eval lib
+    evaluator = get_evaluator(
+        model,
+        base_model,
+        benchmark,
+        judge_model,
+        output_dir,
+        max_workers,
+        taxonomy_path,
+        branch,
+        base_branch,
+        few_shots,
+        batch_size,
+        sdg_path,
+    )
+
+    if benchmark == "mt_bench":
+        # TODO: Replace temp Popen hack with serving library calls. Current library doesn't support server-model-name.
```

Comment:
Noting this is a WIP here: https://github.com/instructlab/instructlab/compare/main...danmcp:instructlab:evalandvllm?expand=1